src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -663,4 +663,33 @@ internal unsafe struct TailCallTls
         public PortableTailCallFrame* Frame;
         public IntPtr ArgBuffer;
     }
+
+    // See src/vm/argslot.h
+    internal static class ArgSlot
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe byte* EndianessFixup(long* pArg, uint cbSize)
+        {
+#if BIGENDIAN
```

Comment:
I would use `if (!BitConverter.IsLittleEndian)` if I were you. In the BCL, the big-endian code gets trimmed away on little-endian platforms on compilation. Here's how ILSpy decompiles `BinaryPrimitives.ReadSingleBigEndian` for example:

![image](https://user-images.githubusercontent.com/12659251/171749565-88d6fd4e-5075-4dac-b7a1-52775a76b670.png)

Reply:
That's interesting, just out of curiosity, why isn't the whole branch trimmed out? 🤔

Reply:
I guess it is important for the trimmer for some reason to keep referencing the field. It will almost certainly be optimized away by the JIT.

> Also, I figured using a directive would be better to minimize the IL size for the method?

I found 16 occurences of `#if !?BIGENDIAN` and 236 occurences of `if \(!?BitConverter.IsLittleEndian\)` in non-test C# code. The overhead of `ldsfld BitConverter.IsLittleEndian` and a branch is 7 bytes (plus a fixed cost for the field's metadata reference). 😅